phoenix-core/src/main/java/org/apache/phoenix/compile/SubselectRewriter.java

```diff
@@ -202,14 +204,30 @@ private SelectStatement flatten(SelectStatement select, SelectStatement subselec
                 }
             }
         }
-        
+        OffsetNode offset = select.getOffset();
+        if (offset != null) {
+            if (offsetRewrite == null) {
+                offsetRewrite = offset;
+            } else {
+                Integer offsetValue = OffsetCompiler.compile(null, select);
+                Integer offsetValueSubselect = OffsetCompiler.compile(null, subselect);
+                if (offsetValue != null && offsetValueSubselect != null) {
+                    offsetRewrite = offsetValue < offsetValueSubselect ? offset : offsetRewrite;
+                } else {
+                    return select;
+                }
+            }
+        }
+
```

Comment:
Thanks for pointing out. I thought a little and have updated the logic to 
 if (offsetRewrite != null || (limitRewrite != null & offset != null)) {
            return select;
        } else {
            offsetRewrite = offset;
        }
Let me know if any optimization is possible.